cmd/kubeapps-apis/plugins/resources/v1alpha1/server.go

```diff
@@ -192,14 +194,16 @@ func (s *Server) GetResources(r *v1alpha1.GetResourcesRequest, stream v1alpha1.R
 		}
 		gvk := groupVersion.WithKind(ref.Kind)
 
-		gvr, err := s.kindToResource(s.restMapper, gvk)
+		// We need to get or watch a different endpoint depending on
+		// the scope of the resource (namespaced or not).
+		gvr, scopeName, err := s.kindToResource(s.restMapper, gvk)
 		if err != nil {
 			return status.Errorf(codes.Internal, "unable to map group-kind %v to resource: %s", gvk.GroupKind(), err.Error())
 		}
 
 		if !r.GetWatch() {
 			var resource interface{}
-			if ref.Namespace != "" {
+			if scopeName == meta.RESTScopeNameNamespace {
```

Comment:
Question: I guess `scopeName` refers to the namespace of the resource, but what 
 does `meta.RESTScopeNameNamespace` stand for?

Reply:
`scopeName` is either `meta.RESTScopeNameNamespace` or `meta.RESTScopeNameRoot`, both of which are [defined in the meta package](https://pkg.go.dev/k8s.io/apimachinery@v0.22.4/pkg/api/meta#RESTScopeName) as string constants corresponding to "namespace" and "root" respectively.